src/columnstore_handler.cpp

```diff
@@ -129,24 +161,29 @@ void columnstore_relation_set_new_filenode(Relation rel, const RelFileNode *newr
                                            TransactionId *freezeXid, MultiXactId *minmulti) {
 #endif
     HeapTuple tp = SearchSysCache1(RELOID, ObjectIdGetDatum(rel->rd_id));
-    if (!HeapTupleIsValid(tp)) {
-        TupleDesc desc = RelationGetDescr(rel);
-        for (int i = 0; i < desc->natts; i++) {
-            Form_pg_attribute attr = &desc->attrs[i];
-            auto duck_type = pgduckdb::ConvertPostgresToDuckColumnType(attr);
-            if (duck_type.id() == duckdb::LogicalTypeId::USER) {
-                elog(ERROR, "column \"%s\" has unsupported type", NameStr(attr->attname));
-            }
-            if (attr->attgenerated) {
-                elog(ERROR, "unsupported generated column \"%s\"", NameStr(attr->attname));
-            }
-        }
-
-        duckdb::Columnstore::CreateTable(rel->rd_id);
-    } else {
+    if (HeapTupleIsValid(tp)) {
         ReleaseSysCache(tp);
         duckdb::Columnstore::TruncateTable(rel->rd_id);
     }
+
+    TupleDesc desc = RelationGetDescr(rel);
+    for (int i = 0; i < desc->natts; i++) {
+        // Check whether column type is acceptable.
+        Form_pg_attribute attr = &desc->attrs[i];
+        auto duck_type = pgduckdb::ConvertPostgresToDuckColumnType(attr);
+        if (duck_type.id() == duckdb::LogicalTypeId::USER) {
+            const auto type_info = duck_type.ToString();
+            elog(ERROR, "column \"%s\" has unsupported user type: %s", NameStr(attr->attname), type_info.data());
```

Comment:
Don't you need to update [test/expected/unsupported/types.out](https://github.com/Mooncake-Labs/pg_mooncake/pull/49/files#diff-cce7d693b39c1c028b91551ff96eee428dfd9727b0e645af660a9fe7be54e324) for this code change?